viewer/event_db.cpp

```diff
@@ -44,7 +55,24 @@ std::vector<EventRef> EventDB::get_data(HT_TimestampNs start_ts,
     }    
     else
     {
-        _query_event_klass(response, _events[query.klass_id], start_ts, stop_ts); 
+        if (_events.find(query.klass_id) == _events.end())
+            return response;
+
+        std::vector<parser::Event>::iterator first_event;
+        std::vector<parser::Event>::iterator last_event;
+        _get_range_of_events(_events[query.klass_id], start_ts, stop_ts, first_event, last_event);
+        --last_event;
+
+        if (_cache.range_in_cache(first_event->get_timestamp(), last_event->get_timestamp(), query.klass_id))
```

Comment:
Does the `_get_range_of_events` guarantee to return valid range? Can it return `(end(), end())`?

Reply:
I think this scenario is a good candidate for unit test

Reply:
_get_range_of_events should compute first_event and last_event such that the range [first_event, last_event) contains all the events between start_ts and stop_ts. 
I've changed the code such that _get_range_of_events is only called when start_st <= stop_ts. I'm not sure if _get_range_of_events should  return TRUE or FALSE if the range exists or not. I've modified the code so that all this checks are done outside of the method.